src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.Windows.cs

```diff
@@ -236,6 +236,16 @@ private bool CreateThread(GCHandle thisThreadHandle)
             return true;
         }
 
+        /// <summary>
+        /// This an entry point for managed threads created by applicatoin
+        /// </summary>
+        [NativeCallable(CallingConvention = CallingConvention.StdCall)]
+        private static uint ThreadEntryPoint(IntPtr parameter)
```

Comment:
It's kind of unfortunate to introduce an extra stack frame in runtime just to avoid `#if`s in the code.